### PR TITLE
Add expect btach with response

### PR DESCRIPTION
**Pull request by alonSadan**
**What this PR does / why we need it**:

-   tests, console: add RunCommandWithResponse
    
    Sometimes it could be useful for debugging,
    to get the Expecter response when running a command against vmi console.
    In this commit two new functions were added to allow running the
    desired command and getting back the Expecter response in addition
    to the error.

-   tests, ping: add response to ping error
    
    The logs returning from the vm performing the ping, could
    be helpful when debugging failed ping tests.
    This commit adds the Expecter response to PIngFromVmConsole, to
    get those logs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by alonSadan** [comment]
now the xfail will show on failure: 

```
[XFAIL] suspected cloud-init issue: https://github.com/kubevirt/kubevirt/issues/4642, failure: Timed out after 15.001s.
    Expected success, but got an error:
        <*multierr.multiError | 0xc002303220>: {
            copyNeeded: {v: 0},
            errors: [
                {
                    s: "Failed to ping VMI testvmi-f6xfl, error: Failed to run [ping wrongping_1 -c 1 -w 5] at VMI testvmi-f6xfl, error: ping wrongping_1 -c 1 -w 5 failed, response: [{1 \r\n[root@testvmi-f6xfl fedora]#  [#  # ]} {3 ping wrongping_1 -c 1 -w 5\r\nping: wrongping_1: Name or service not known\r\n[root@testvmi-f6xfl fedora]#  [#  # ]} {5 echo $?\r\n2\r\n[root@testvmi-f6xfl fedora]#  [\n2\r\n[root@testvmi-f6xfl fedora]#  # ]}]",
                },
                {
                    s: "Failed to ping VMI testvmi-md8ff, error: Failed to run [ping worngping_2 -c 1 -w 5] at VMI testvmi-md8ff, error: ping worngping_2 -c 1 -w 5 failed, response: [{1 \r\n[root@testvmi-md8ff fedora]#  [#  # ]} {3 ping worngping_2 -c 1 -w 5\r\nping: worngping_2: Name or service not known\r\n[root@testvmi-md8ff fedora]#  [#  # ]} {5 echo $?\r\n2\r\n[root@testvmi-md8ff fedora]#  [\n2\r\n[root@testvmi-md8ff fedora]#  # ]}]",
                },
            ],
        }
        Failed to ping VMI testvmi-f6xfl, error: Failed to run [ping wrongping_1 -c 1 -w 5] at VMI testvmi-f6xfl, error: ping wrongping_1 -c 1 -w 5 failed, response: [{1 
        [root@testvmi-f6xfl fedora]#  [#  # ]} {3 ping wrongping_1 -c 1 -w 5
        ping: wrongping_1: Name or service not known
        [root@testvmi-f6xfl fedora]#  [#  # ]} {5 echo $?
        2
        [root@testvmi-f6xfl fedora]#  [
        2
        [root@testvmi-f6xfl fedora]#  # ]}]; Failed to ping VMI testvmi-md8ff, error: Failed to run [ping worngping_2 -c 1 -w 5] at VMI testvmi-md8ff, error: ping worngping_2 -c 1 -w 5 failed, response: [{1 
        [root@testvmi-md8ff fedora]#  [#  # ]} {3 ping worngping_2 -c 1 -w 5
        ping: worngping_2: Name or service not known
        [root@testvmi-md8ff fedora]#  [#  # ]} {5 echo $?
        2
        [root@testvmi-md8ff fedora]#  [
        2
        [root@testvmi-md8ff fedora]#  # ]}]

```

**Comment by alonSadan** [comment]
Removed to xfail, now the error should include also the boot log

File: tests/vmi_multus_test.go [hunk 1]
```diff
@@ -853,16 +856,44 @@ var _ = Describe("[Serial]SRIOV", func() {
...
+				err1 := multierr.Combine(libnet.PingFromVMConsole(vmi1, cidrToIP(cidrB)), libnet.PingFromVMConsole(vmi2, cidrToIP(cidrA)),
+					libnet.PingFromVMConsole(vmi1, cidrToIP(cidrDebug)), libnet.PingFromVMConsole(vmi2, cidrToIP(cidrDebug)))
+
+				_, err2 := RunOnNode("sriov-worker", fmt.Sprintf("ip addr del %s dev enp4s0f0", "192.168.0.100"))
+				err = multierr.Append(err1, err2)
+
+				vm1respIPRoute, _ := console.RunCommandWithResponse(vmi1, "ip route", maxCommandTimeout)
+				vm1respIPLink, _ := console.RunCommandWithResponse(vmi1, "ip link", maxCommandTimeout)
+				vm1respIPAddress, _ := console.RunCommandWithResponse(vmi1, "ip address", maxCommandTimeout)
+				vm1respIPNeigh, _ := console.RunCommandWithResponse(vmi1, "ip neigh", maxCommandTimeout)
```
Comment (ormergi) [inline tests/vmi_multus_test.go:880]: Could you please run those before the ping command
we better check what happen before the ping as well

**Comment by alonSadan** [comment]
pull-kubevirt-e2e-kind-1.17-sriov

File: tests/vmi_multus_test.go (hunk 1 above)
Comment (alonSadan) [reply tests/vmi_multus_test.go:874]: Done

**Comment by alonSadan** [comment]
This  PR was used for debugging and it is no longer needed.

**Comment by alonSadan** [comment]
> This PR was used for debugging and it is no longer needed.

The issue that was debugged still exists. reopen.

**Comment by alonSadan** [comment]
This was a debugging PR and it no longer needed.

**Pull request closed without merging**